streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -335,12 +335,23 @@
 
     /**
      * Count number of messages of this stream by key without a window basis, and hence
+     * return a ever updating counting table.
```

Comment:
While we're at it, I'd strongly suggest to clarify the behavior of `count()` for both KTable and KStream.

My argument is that most users would consider a `count()` function to accept a predicate and return a single number, like Scala's `count()`:

```scala
scala> "kafka".count(_ == 'k')
res1: Int = 2
```

In the DSL however, we are returning a `KTable<K, Long>`.  My recommendation would be to 1. add 1-2 clarifying sentences to the docstring and, if it makes sense, 2. add a simple example to the docstring.  (And do this for both KTable and KStream)

What do you think, @ijuma?